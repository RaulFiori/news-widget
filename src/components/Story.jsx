import React from 'react';
import moment from 'moment';
import styled from 'styled-components';
import { storyShape } from '../prop-types/news-types';

const Container = styled.div`
  border-bottom: 2px solid #a4a4a4;
  padding: 16px;
  padding-left: 0px;
`;

const Title = styled.a`
  text-align: left;
  font-family: 'Roboto', sans-serif;
  letter-spacing: 0;
  color: #4e4e4e;
  opacity: 1;
  text-decoration: none;
`;

const Caption = styled.span`
  color: #a4a4a4;
  font-family: 'Roboto', sans-serif;
  letter-spacing: 0;
  margin-right: 20px;
`;

const Chip = styled.div`
  background: #d8d8e4 0% 0% no-repeat padding-box;
  border-radius: 4px;
  opacity: 1;
  padding: 4px;
  color: white;
`;

const Story = props => {
  const {
    story: {
      title,
      publishedAt,
      url,
      source: { name }
    }
  } = props;
  return (
    <Container>
      <Title href={url} rel="noopener noreferrer" target="_blank">
        {title}
      </Title>
      <div style={{ display: 'flex', alignItems: 'center', marginTop: '16px' }}>
        <Caption>{moment(publishedAt).format('DD/MM/YYYY')}</Caption>
        <Chip>{name}</Chip>
      </div>
    </Container>
  );
};

Story.propTypes = {
  story: storyShape.isRequired
};

export default Story;
