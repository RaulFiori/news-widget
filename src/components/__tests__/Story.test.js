import React from 'react';
import renderer from 'react-test-renderer';
import Story from '../Story';

const mockStory = {
  title: 'Some news',
  publishedAt: '2019-08-07T16:30:00Z',
  url: 'some-site.com.br',
  source: {
    name: 'news site'
  }
};

describe('Story Component', () => {
  test('should render correctly', () => {
    const element = renderer.create(<Story story={mockStory} />).toJSON();
    expect(element).toMatchSnapshot();
  });
});
