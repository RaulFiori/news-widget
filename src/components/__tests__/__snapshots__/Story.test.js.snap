// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Component should render correctly 1`] = `
<div
  className="sc-bdVaJa ihSyXN"
>
  <a
    className="sc-bwzfXH nGdMX"
    href="some-site.com.br"
    rel="noopener noreferrer"
    target="_blank"
  >
    Some news
  </a>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "marginTop": "16px",
      }
    }
  >
    <span
      className="sc-htpNat dRQqp"
    >
      07/08/2019
    </span>
    <div
      className="sc-bxivhb cLWSfx"
    >
      news site
    </div>
  </div>
</div>
`;
